src/plugins/intel_cpu/src/nodes/mkldnn_matmul_node.cpp

```diff
@@ -22,9 +22,357 @@
+
+        this->postamble();
+
+        load_emitter->emit_data();
+        store_emitter->emit_data();
+
+        for (auto& inj : eltwise_injectors)
+            inj->prepare_table();
+    }
+
+private:
+    using Vmm = typename conditional3<isa == x64::sse41, Xbyak::Xmm, isa == x64::avx2, Xbyak::Ymm, Xbyak::Zmm>::type;
+    const int vlen = cpu_isa_traits<isa>::vlen;
+    const int vec_step = vlen / sizeof(float);
+    int amount_full;
+    int amount_tail;
+    int unroll_dst_count = 1;
+
+    Vmm get_src_vmm(const int idx) {
+        return Vmm(1 + unroll_dst_count + idx);
+    }
+
+    Vmm get_dst_vmm(const int idx = 0) {
+        return Vmm(1 + idx);
+    }
+
+    Xmm get_aux_xmm(const int idx) {
+        return Xmm(idx);
+    }
+
+    Xbyak::Reg64 reg_src_0 = r8;
+    Xbyak::Reg64 reg_src_1 = r9;
+    Xbyak::Reg64 reg_dst = r10;
+    Xbyak::Reg64 reg_src_aux_0 = r11;
+    Xbyak::Reg64 reg_src_aux_1 = r12;
+    Xbyak::Reg64 reg_dst_aux = r13; // only for 2nd algorithm
+
+    // indexes
+    Xbyak::Reg64 batch = r14;
+    Xbyak::Reg64 m = r15;
+    Xbyak::Reg64 k = rax;
+    Xbyak::Reg64 n = rbx;
+    Xbyak::Reg64 n_inside = rdx;
+
+    Xbyak::Reg64 reg_params = abi_param1; // RDI | RCX
+
+    // loaders and stores
+    Xbyak::Reg64 reg_load_store_mask = rsi;
+    Xbyak::Reg64 reg_load_table = rbp;
+
+    Vmm vmm_zero = Vmm(0);
+
+    std::unique_ptr<jit_load_emitter> load_emitter = nullptr;
+    std::unique_ptr<jit_store_emitter> store_emitter = nullptr;
+
+    std::vector<size_t> load_pool_gpr_idxs;
+    std::vector<size_t> store_pool_gpr_idxs;
+    std::vector<size_t> store_pool_vec_idxs;
+
+    std::vector<std::shared_ptr<jit_uni_eltwise_injector_f32<isa>>> eltwise_injectors;
+
+    inline void body() {
+        Xbyak::Label label_m;
+        Xbyak::Label label_m_end;
+
+        mov(m, 0);
+        L(label_m); {
+            cmp(m, jcp_.m);
+            je(label_m_end, T_NEAR);
+
+            if (jcp_.scalar_product) {
+                optimized_body_loop();
+            } else {
+                body_loop();
+            }
+
+            add(reg_src_0, jcp_.k * sizeof(float));
+            add(m, 1);
+            jmp(label_m, T_NEAR);
+        }
+        L(label_m_end);
+    }
+
+    // common execution : broadcast(a) * vmm_b
+    inline void body_loop() {
+        Xbyak::Label label_k;
+        Xbyak::Label label_k_end;
+
+        unroll_dst_count = amount_full + static_cast<int>(amount_tail != 0);
+
+        mov(reg_src_aux_0, reg_src_0);
+        mov(reg_src_aux_1, reg_src_1);
+        for (int i = 0; i < unroll_dst_count; ++i)
+            uni_vpxor(get_dst_vmm(i), get_dst_vmm(i), get_dst_vmm(i));
+
+        mov(k, 0);
+        L(label_k); {
+            cmp(k, jcp_.k);
+            je(label_k_end, T_NEAR);
+
+            uni_vbroadcastss(get_src_vmm(0), ptr[reg_src_aux_0]);  // src0
+
+            for (int i = 0; i < amount_full; ++i) {
+                load_ptr(get_src_vmm(1), ptr[reg_src_aux_1 + i * vlen]);  // src1
```

Comment:
1. `load_prt` is a confusing name. It says that you are trying to load ptr, not the data.
2. Why cannot you use load emitter here?

Reply:
Discussed offline. c09f226f20eab6926a045752e2fed7434931a152